plugins/outputs/datadog/datadog.go

```diff
@@ -168,6 +169,9 @@ func (d *Datadog) Write(metrics []telegraf.Metric) error {
 	defer resp.Body.Close()
 
 	if resp.StatusCode < 200 || resp.StatusCode > 209 {
+		// err can be ignored
+		body, _ := io.ReadAll(resp.Body)
+		d.Log.Debugf("Response from datadog api %s", string(body))
 		return fmt.Errorf("received bad status code, %d", resp.StatusCode)
```

Comment:
if we are already returning an error, is there a reason to not add the body to that error message? Something like:

```suggestion
		return fmt.Errorf("received bad status code, %d: %s", resp.StatusCode, string(body))
```

Reply:
Although that is a good option, I left it out from the error message considering the body could be large string which would also increase the error message size. If that is acceptable, then we can make that change.

Reply:
Do you have an example of a really long message you could share?
Where we say bad status code in other places, we have been slowly adding the body in as well, so I think it should be ok, but what a long message looks like might help. 

Thanks!

Reply:
Official documents dont have long messages, but we have a usecase where we sort of validate the datadog json body at a proxy and forward it to the cloud. Our validation could return a json which is around 20 lines. But given that it is a specific usecase, I feel its acceptable to use what you suggest. 
Thanks!